### ISSUE TITLE
libp2p-host doesn't build

### ISSUE THREAD

**Issue by faddat** [libp2p/go-libp2p#229]
![image](https://user-images.githubusercontent.com/7142025/30254356-5c7ce296-9665-11e7-8689-56452ef84f2e.png)

There are two possiblities here and I haven't investigated yet-- the first is that the example app is a very bare example and won't build till fleshed out. The second is that I've found a bug or some breakage from code churn. I'm not certain which it is and need to head to sleep, so figured I would post it regardless.

**Comment by Stebalien** [comment]
TL;DR: Run `make deps` in the root of the repository and try again.

We're working on pulling through some breaking changes so things are a bit broken at the moment if you use `go get` for dep management. This is why we use a tool we made called `gx` as our canonical dependency manger. It uses content addressing to address the dependencies (really, it just stores the deps in IPFS) so it tends not to have issues like this.

If you're wondering how this relates to the solution above, `make deps` will:

1. Install `gx` using `go get`.
2. Fetch the dependencies using `gx`.
3. Rewrite the dependencies in the go code to point to the dependencies fetched through `gx`.

---

In the future, we'd like to make it possible to fetch deps managed by `gx` using `go get` so everything "just works" (you'll never end up with the wrong dependency versions). We've (well, @lgierth and @whyrusleeping) recently figured out a way to do this that we think will work but have yet to implement it.
